File: Solutions/844. Backspace String Compare/844. Backspace String Compare.go
func backspaceCompare(s string, t string) bool {
    i,j,back:=len(s)-1,len(t)-1,0
    for true{
        back=0
        for i>=0 && (back>0 || s[i]=='#'){
            if s[i]=='#'{
                back+=1
            }else{
                back-=1
            }
            i-=1
        }
        back=0
        for j>=0 && (back>0 || t[j]=='#'){
            if t[j]=='#'{
                back+=1
            }else{
                back-=1
            }
            j-=1
        }
        if i>=0 && j>=0 && s[i]==t[j]{
            i-=1;j-=1
        }else{
            break
        }
    }
    return i==-1 && j==-1
}

File: Solutions/844. Backspace String Compare/844. Backspace String Compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        int i=s.size()-1,j=t.size()-1,back;
        while(true){
            back=0;
            while(i>=0 and (back>0 or s[i]=='#')){
                back+=s[i--]=='#'?1:-1;
            }
            back=0;
            while(j>=0 and (back>0 or t[j]=='#')){
                back+=t[j--]=='#'?1:-1;
            }
            if(i>=0 and j>=0 and s[i]==t[j]){
                --i;
                --j;
            }else{
                break;
            }
        }
        return i==-1 and j==-1;
    }
};